src/psyclone/domain/common/transformations/kernel_module_inline_trans.py

```diff
@@ -319,21 +317,111 @@ def _get_psyir_to_inline(node):
                     f"{[rout.name for rout in routines]}. TODO #924")
         return (caller_name, routines[0])
 
+    @staticmethod
+    def _rm_imported_routine_symbol(symbol: Symbol,
+                                    schedule: Routine,
+                                    table: SymbolTable):
+        '''
+        If the named symbol is in the supplied table (or an ancestor) *and* is
+        an import then it is removed. If the Container from which it was being
+        imported no longer has any imports associated with it then the
+        ContainerSymbol is also removed.
+
```

Comment:
I have some problems with this method but maybe not worth solving if it needs to be replaced, but I think I would bring the TODO 2846 from inside the implementation to the docstring saying "removing an imported routine symbol is a dangerous operation, one alternative is renaming the symbol of the inlined routine"

Reply:
Yes, done and reworded.